librz/analysis/esil/esil_trace.c

```diff
@@ -368,86 +368,61 @@ RZ_API void rz_analysis_esil_trace_restore(RzAnalysisEsil *esil, int idx) {
 	ht_up_foreach(trace->memory, restore_memory_cb, esil);
 }
 
-static void print_instruction_trace(RzILTraceInstruction *instruction, int idx) {
+static void print_instruction_ops(RzILTraceInstruction *instruction, int idx, RzILTraceInsOp focus) {
+	bool reg = focus == RZ_IL_TRACE_INS_HAS_REG_R || focus == RZ_IL_TRACE_INS_HAS_REG_W;
```

Comment:
```suggestion
	bool reg = focus & (RZ_IL_TRACE_INS_HAS_REG_R | RZ_IL_TRACE_INS_HAS_REG_W);
```
`RzILTraceInsOp` is a bitfield!

Reply:
Yes but it can be used as an enum just fine and I am using it as an enum. If I use it as a bitfield, it's no longer proper to have the `focus` parameter be of type `RzILTraceInsOp`.

Reply:
I agree with @kazarmy